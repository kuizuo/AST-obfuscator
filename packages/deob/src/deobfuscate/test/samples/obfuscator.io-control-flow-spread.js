function hi() {
  const _0x5e8644 = _0x5c1c
  const _0xa7728a = {
    UcfbR(_0x5a9ca0, ..._0x56527a) {
      return _0x5a9ca0(..._0x56527a)
    },
  }
  return m.functionExpression(undefined, captures[_0x5e8644(0x1)](m[_0x5e8644(0x3)]), m[_0x5e8644(0x0)](_0xa7728a[_0x5e8644(0x2)](body, ...captures[_0x5e8644(0x1)](_0x56f63c => m[_0x5e8644(0x3)](m.fromCapture(_0x56f63c))), ...arr)))
}
function _0x580f() {
  const _0x3034f3 = [
    'blockStatement',
    'map',
    'UcfbR',
    'identifier',
  ]
  _0x580f = function () {
    return _0x3034f3
  }
  return _0x580f()
}
function _0x5c1c(_0x580f21, _0x5c1c8f) {
  const _0x50ceb3 = _0x580f()
  _0x5c1c = function (_0xc280f8, _0x2a8aa7) {
    _0xc280f8 = _0xc280f8 - 0x0
    const _0x16019b = _0x50ceb3[_0xc280f8]
    return _0x16019b
  }
  return _0x5c1c(_0x580f21, _0x5c1c8f)
}
hi()
