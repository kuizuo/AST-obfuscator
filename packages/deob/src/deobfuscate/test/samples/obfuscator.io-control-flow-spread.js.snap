function hi() {
  return m.functionExpression(undefined, captures.map(m.identifier), m.blockStatement(body(...captures.map(_0x56f63c => m.identifier(m.fromCapture(_0x56f63c))), ...arr)));
}
hi();